packages/compass-e2e-tests/helpers/test-runner-global-fixtures.ts

```diff
@@ -0,0 +1,183 @@
+import gunzip from './gunzip';
+import path from 'path';
+import fs from 'fs';
+import {
+  DEFAULT_CONNECTION_PORTS,
+  DEFAULT_CONNECTIONS_SERVER_INFO,
+  DISABLE_START_STOP,
+  E2E_WORKSPACE_PATH,
+  LOG_PATH,
+  SKIP_COMPASS_DESKTOP_COMPILE,
+  SKIP_NATIVE_MODULE_REBUILD,
+  TEST_COMPASS_DESKTOP_PACKAGED_APP,
+  TEST_COMPASS_WEB,
+} from './test-runner-context';
+import Debug from 'debug';
+import type { MongoCluster } from '@mongodb-js/compass-test-server';
+import { startTestServer } from '@mongodb-js/compass-test-server';
+import crossSpawn from 'cross-spawn';
+import kill from 'tree-kill';
+import { MongoClient } from 'mongodb';
+import { isEnterprise } from 'mongodb-build-info';
+import {
+  buildCompass,
+  compileCompassAssets,
+  rebuildNativeModules,
+  removeUserDataDir,
+} from './compass';
+
+const debug = Debug('compass-e2e-tests:mocha-global-fixtures');
+
+const wait = (ms: number) =>
+  new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
+
+const cleanupFns: (() => Promise<void> | void)[] = [];
+
+const servers: MongoCluster[] = [];
+
+/**
+ * Main hook that does all the main pre-setup before running tests:
+ *  - Unpacks the fixtures
+ *  - Starts MongoDB servers for every default connection defined in
+ *    DEFAULT_CONNECTIONS
+ *  - Starts compass-web sandbox
+ *  - Updates server metadata that will be used by tests
+ *  - Compiles the desktop app
+ */
+export async function mochaGlobalSetup(this: Mocha.Runner) {
+  debug('Unzipping fixtures...');
+  await gunzip(path.join(E2E_WORKSPACE_PATH, 'fixtures', '*.gz'));
+
+  debug('X DISPLAY', process.env.DISPLAY);
+
+  if (!DISABLE_START_STOP) {
+    debug('Starting MongoDB servers');
+
+    for (const port of DEFAULT_CONNECTION_PORTS) {
+      const server = await startTestServer({
+        args: ['--port', port],
+        version:
+          process.env.MONGODB_VERSION ?? process.env.MONGODB_RUNNER_VERSION,
+      });
+      servers.push(server);
+      cleanupFns.push(() => {
+        debug('Stopping server at port %s', port);
+        return server.close();
+      });
+    }
+
+    if (TEST_COMPASS_WEB) {
+      debug('Starting Compass Web');
+      const compassWeb = spawnCompassWeb();
+      cleanupFns.push(() => {
+        if (compassWeb.pid) {
+          debug(`Killing compass-web [${compassWeb.pid}]`);
+          kill(compassWeb.pid, 'SIGINT');
+        } else {
+          debug('No pid for compass-web');
+        }
+      });
+      await waitForCompassWebToBeReady();
+    }
+  }
+
+  debug('Getting mongodb server info');
+  await updateMongoDBServerInfo();
+
+  try {
+    debug('Clearing out past logs');
+    fs.rmdirSync(LOG_PATH, { recursive: true });
+  } catch (e) {
+    debug('.log dir already removed');
+  }
+
+  fs.mkdirSync(LOG_PATH, { recursive: true });
+
+  if (TEST_COMPASS_DESKTOP_PACKAGED_APP) {
+    debug('Building Compass before running the tests ...');
+    await buildCompass();
```

Comment:
Some of these probably can be just moved here directly as they are just running scripts with crossSpawn, then it's maybe easier to read through the compass.ts file